### DISCUSSION TITLE
code block implicitly closed when its parent container is closed?

### DISCUSSION THREAD

**Discussion by uvtc** [jgm/djot#267] [General]
I just noticed [in the docs for code blocks](https://htmlpreview.github.io/?https://github.com/jgm/djot/blob/master/doc/syntax.html#code-block) that this

~~~~~~
> ```
> code in a
> block quote

Paragraph.
~~~~~~

is valid syntax. I was expecting it to instead be:

~~~~~~
> ```
> code in a
> block quote
> ```

Paragraph.
~~~~~~

The former (without the closing ```` ``` ````) looks strange to me. What is the rationale for the implicit closing of the code block?

I see that I can also start a div inside a blockquote and it too doesn't require the closing `:::`:

~~~~~~
Foo

> ::: warning
> implictly-closed div
> block in this blockquote!

Bar
~~~~~~

And I see that I can also do something similar, but with a div inside a div:

~~~~~~
Foo

::: warning-1
this is a warning
one.

2nd para.

::: warning-2
This is a warning
two, within the
warning one.
:::

Bar
~~~~~~

where that "Bar" is outside both divs... But, does that look correct though? warning-1 hasn't closed yet, so I'd expect Bar to still be inside it.

**Comment by jgm** [comment]
This is syntax that works, but it's not what I'd recommend writing.  For clarity, include the closing ticks or colons.  But we need to tell the parser what to do in the case where the outer container ends and the inner one isn't closed yet -- closing implicitly seems the best option.
